src/CSharp/CodeCracker/Refactoring/IntroduceFieldFromConstructorAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class IntroduceFieldFromConstructorAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Title = "Consider introduce field for constructor parameters.";
+        internal const string MessageFormat = "Introduce a field for parameter: {0}";
+        internal const string Category = SupportedCategories.Refactoring;
+        const string Description = "Consider introduce field for constructor parameters.";
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId.IntroduceFieldFromConstructor.ToDiagnosticId(),
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Hidden,
+            isEnabledByDefault: true,
+            description: Description,
+            helpLink: HelpLink.ForDiagnostic(DiagnosticId.TaskNameAsync));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeConstructor, SyntaxKind.ConstructorDeclaration);
+
+        private static void AnalyzeConstructor(SyntaxNodeAnalysisContext context)
+        {
+            var constructorMethod = (ConstructorDeclarationSyntax)context.Node;
+            var parameters = constructorMethod.ParameterList.Parameters;
+
+            if (constructorMethod.Body == null) return;
+            if (constructorMethod?.Body.Statements.Count == 0 && parameters.Count > 0)
+            {
+                foreach (var par in parameters)
+                {
+                    var errorMessage = par.Identifier.Text;
+                    var diag = Diagnostic.Create(Rule, constructorMethod.GetLocation(), errorMessage);
+                    context.ReportDiagnostic(diag);
+                }
+                return;
+            }
+
+            foreach (var par in parameters)
+            {
+                var foundAssignment = false;
+                foreach (var statement in constructorMethod.Body.Statements)
+                {
+                    if (statement.IsKind(SyntaxKind.ExpressionStatement))
+                    {
+                        var expression = statement as ExpressionStatementSyntax;
+                        var assign = expression.Expression as AssignmentExpressionSyntax;
+                        if ((assign?.Left.IsKind(SyntaxKind.SimpleMemberAccessExpression) ?? false) && (assign?.Right.IsKind(SyntaxKind.IdentifierName) ?? false))
+                        {
+                            var right = assign.Right as IdentifierNameSyntax;
+                            if (right.Identifier.Text == par.Identifier.Text)
+                            {
+                                foundAssignment = true;
+                                break;
+                            }
+                        }
+                    }
+                }
```

Comment:
This is not going to work if the assignment is inside an if statement.
Maybe you should use a data flow analysis?